rednotebook/gui/clouds.py

```diff
@@ -157,9 +151,27 @@ def _get_cloud_body(self, cloud_words):
             self.link_index += 1
         return '\n'.join(html_elements)
 
+    @staticmethod
+    def _get_collated_word(word_and_freq):
+        word, freq = word_and_freq
+        return locale.strxfrm(word)
+
+    @staticmethod
+    def _frequency(word_and_freq):
+        word, freq = word_and_freq
+        return freq
+
     def _get_tags_for_cloud(self, tag_count_dict, ignores):
-        return [(tag, freq) for (tag, freq) in tag_count_dict
-                if not any(pattern.match(tag) for pattern in ignores)]
+        tag_display_limit = self.journal.config.read('cloudMaxTags')
+        if tag_display_limit == 0:
+            return []
+
+        tags_and_frequencies = [(tag, freq) for (tag, freq) in tag_count_dict
+                                if not any(pattern.match(tag) for pattern in ignores)]
+
+        # Sort by the frequency of occurrence
+        tags_and_frequencies.sort(key=self._frequency, reverse=True)
+        return tags_and_frequencies[:tag_display_limit]
```

Comment:
The tags should also be sorted alphabetically.

Reply:
It might be a good idea to factor out the filtering and sorting of tags and words into a function or a static method. This function or method can have `get_collated_word` and `frequency` functions as local functions.

Reply:
Yeah, it's cleaner now.